### PR TITLE
Feature/entity disabled

### LINKED ISSUE #19: Add an "entity_disabled" argument to each created dataclass
Defaults to False

Like type: https://github.com/epics-containers/ibek/blob/master/ibek/ioc.py#L102

If set then the render code should ignore this element

### PR DESCRIPTION
**Pull request by BenBradnick**
This is to fix #19.

Currently mypy complains because the base class Entity has no attribute entity_disabled - but if I add a type hint to this class then other unit tests fail due to an argument with a default value then being defined before a non-default argument.

I could ignore the check on that line but I was wondering if there was a real fix?

### CODE REVIEW

**Comment by thomascobb** [comment]
If you put it in the baseclass too without the initial value, but remove the dataclass decorator, does that work?

**Comment by BenBradnick** [comment]
Hi @thomascobb can you take another look now that CI looks happy again?

File: setup.cfg [hunk 1]
```diff
@@ -22,11 +22,35 @@ install_requires =
...
+    # Pinning black stops us having to allow pre-releases globally
+    black==21.9b0
+    # Flake version 4.0 and 4.0.1 do not like pytest: https://github.com/tholo/pytest-flake8/issues/81
+    flake8==3.9.2
+    # Pins to make lockfile usable on multiple Python versions and platforms
+    mypy
+    atomicwrites
+    typing-extensions
+    importlib-metadata
+    typed-ast
```
Comment (thomascobb) [inline setup.cfg:42]: These lines can go, as we now only use the lockfile on one python

File: setup.cfg [hunk 2]
```diff
@@ -22,11 +22,35 @@ install_requires =
...
+    # Test and docs dependencies
+    pytest-cov
+    pytest-mypy
+    pytest-flake8
+    pytest-black
+    flake8-isort
+    isort>=5.0
+    sphinx-rtd-theme
+    sphinx-apischema
+    autopep8==1.5.7
```
Comment (thomascobb) [inline setup.cfg:52]: What's autopep8 used for?

Comment (BenBradnick) [reply setup.cfg:52]: Not sure, guessing someone was previously using it as an alternative to black?

Comment (BenBradnick) [reply setup.cfg]: I have removed it anyway...

**Pull request merged**